charts/agent/templates/_helpers.tpl

```diff
@@ -0,0 +1,194 @@
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper Sysdig Agent image name for module building
+*/}}
+{{- define "agent.image.kmodule" -}}
+    {{- include "agent.imageRegistry" . -}} / {{- .Values.slim.kmoduleImage.repository -}} {{- if .Values.slim.kmoduleImage.digest -}} @ {{- .Values.slim.kmoduleImage.digest -}} {{- else -}} : {{- .Values.image.tag -}} {{- end -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "agent.labels" -}}
+helm.sh/chart: {{ include "agent.chart" . }}
+{{ include "agent.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "agent.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "agent.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Daemonset labels
+*/}}
+{{- define "daemonset.labels" -}}
+  {{- if .Values.daemonset.labels }}
+    {{- $tp := typeOf .Values.daemonset.labels }}
+    {{- if eq $tp "string" }}
+      {{- tpl .Values.daemonset.labels . }}
+    {{- else }}
+      {{- toYaml .Values.daemonset.labels }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{/*
+Use like: {{ include "get_or_fail_if_in_settings" (dict "root" . "key" "<mypath.key>" "setting" "<agent_setting>") }}
+Return the value of key "<mypath.key>" and if "<agent_setting>" is also defined in sysdig.settings.<agent_setting>, and error is thrown
+NOTE: I don't like the error message! Too much information.
+*/}}
+{{- define "get_or_fail_if_in_settings" -}}
+{{- $keyValue := tpl (printf "{{- .Values.%s -}}" .key) .root }}
+{{- if $keyValue -}}
+    {{- if hasKey .root.Values.sysdig.settings .setting }}{{ fail (printf "Value '%s' is also set via .sysdig.settings.%s'." .key .setting) }}{{- end -}}
+    {{- $keyValue -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+The following helper functions are all designed to use global values where
+possible, but accept overrides from the chart values.
+*/}}
+{{- define "agent.accessKey" -}}
+    {{- required "A valid accessKey is required" (.Values.sysdig.accessKey | default .Values.global.sysdig.accessKey) -}}
+{{- end -}}
+
+{{- define "agent.accessKeySecret" -}}
+    {{/*
+    Note: the last default function call is to avoid some weirdness when either
+    argument is nil. If .Values.global.sysdig.accessKeySecret was undefined, the
+    returned empty string does not evaluate to empty on Helm Version:"v3.8.0"
+    */}}
+    {{- .Values.sysdig.existingAccessKeySecret | default .Values.global.sysdig.accessKeySecret | default "" -}}
+{{- end -}}
+
+{{- define "agent.clusterName" -}}
+    {{- $clusterName := include "get_or_fail_if_in_settings" (dict "root" . "key" "clusterName" "setting" "k8s_cluster_name") }}
+    {{- $clusterName | default .Values.global.clusterConfig.name | default "" -}}
+{{- end -}}
+
+{{/*
```

Comment:
This is also a new set of helpers to determine when to automatically use ebpf.

Reply:
This is nice. I don't know how well it plays with "helm template" (i.e. some customers generate the YAMLs and then apply on the cluster). Does it work offline?